packages/grid/x-data-grid/src/hooks/features/columnGrouping/useGridColumnGrouping.ts

```diff
@@ -63,6 +63,8 @@ export const columnGroupsStateInitializer: GridStateInitializer<
   const columnGroupsHeaderStructure = getColumnGroupsHeaderStructure(
     columnFields,
     unwrappedGroupingModel,
+    // @ts-expect-error Move this part to `Pro` package
```

Comment:
To avoid these we had to duplicate a lot of logic in Pro package (by copying this hook and updating the section which handles `pinning` part)
Suggestions are welcome!

Reply:
Why are these comments using `@ts-...`?

Reply:
Because we are trying to access part of the state in MIT package which is typed in the Pro package (column pinning), so the typescript complains, in order to make him happy I have to use `@ts-expect-error`
I guess we are following the same for some other flags too which have to access in a lower package.

Reply:
Ideally, we would use a pipe processor to spread the logic between packages, but I think it's not worth it in this case since it will bring unnecessary complexity 👍🏻

Reply:
Exactly, in this case, and other cases like keyboard navigation for header filters for example, I am thinking of extracting most of the functions from the actual hook (as reusable ones) and having dedicated hooks in each package with the differences added on top of it.

It might be tougher to achieve but a step forward towards https://github.com/mui/mui-x/issues/7358